.github/workflows/ci.yml

```diff
@@ -22,51 +22,89 @@ jobs:
+      rootless-docker: false
+      ryuk-disabled: false
 
-      - name: Set up Go
-        uses: actions/setup-go@v3
-        with:
-          go-version: ${{ matrix.go-version }}
-        id: go
-
-      - name: Check out code into the Go module directory
-        uses: actions/checkout@v3
-
-      - name: modVerify
-        run: go mod verify
-
-      - name: modTidy
-        run: make tools-tidy
-
-      - name: ensure compilation
-        env:
-          GOOS: linux
-        run: go build
+  # The job below is a copy of the job above, but with ryuk disabled.
+  # It's executed in a secondary stage to avoid concurrency issues.
+  test-reaper-off:
+    name: "Test with reaper off"
+    needs: test
+    strategy:
+      matrix:
+        go-version: [1.19.x, 1.x]
+    uses: ./.github/workflows/ci-test-go.yml
+    with:
+      go-version: ${{ matrix.go-version }}
+      platform: "ubuntu-latest"
+      project-directory: "."
+      rootless-docker: false
+      ryuk-disabled: true
 
-      - name: go test
-        # only run tests on linux, there are a number of things that won't allow the tests to run on anything else
-        # many (maybe, all?) images used can only be build on Linux, they don't have Windows in their manifest, and
-        # we can't put Windows Server in "Linux Mode" in Github actions
-        # another, host mode is only available on Linux, and we have tests around that, do we skip them?
-        if: ${{ matrix.platform == 'ubuntu-latest' }}
-        run: |
-          go install gotest.tools/gotestsum@latest
-          make test-unit
+  # The job below is a copy of the job above, but with Docker rootless.
+  # It's executed in a secondary stage to avoid concurrency issues.
+  test-rootless-docker:
+    name: "Test with Rootless Docker"
+    needs: test
+    strategy:
+      matrix:
+        go-version: [1.19.x, 1.x]
+        platform: [ubuntu-latest]
+    uses: ./.github/workflows/ci-test-go.yml
+    with:
+      go-version: ${{ matrix.go-version }}
+      platform: "ubuntu-latest"
+      project-directory: "."
+      rootless-docker: true
+      ryuk-disabled: false
 
-      - name: Run module generator tests
-        if: ${{ matrix.platform == 'ubuntu-latest' }}
-        run: make -C modulegen test-unit
+  test-module-generator:
+    strategy:
+      matrix:
+        go-version: [1.19.x, 1.x]
+        platform: [ubuntu-latest, macos-latest]
+    uses: ./.github/workflows/ci-test-go.yml
+    with:
+      go-version: ${{ matrix.go-version }}
+      platform: ${{ matrix.platform }}
+      project-directory: "modulegen"
+      rootless-docker: false
+      ryuk-disabled: false
 
-      - name: Run checker
-        run: |
-          ./scripts/check_environment.sh
+  test-modules:
+    needs: test
+    strategy:
+      matrix:
+        go-version: [1.19.x, 1.x]
+        platform: [ubuntu-latest, macos-latest]
+        module: [compose, couchbase, k3s, localstack, mysql, neo4j, postgres, pulsar, redis, redpanda, vault]
```

Comment:
I was wondering if we can auto discover the folders under modules to build the matrix automatically

Reply:
This file is autogenerated by the modules generator